devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy

```diff
@@ -0,0 +1,314 @@
+					eventType: "ALERT",
+					name: "secureInclusion",
+					value: "failed",
+					displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description, [0x98: 1, 0x20: 1, 0x70: 1])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x85: 2, 0x70: 1])
+	// log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(basicv1.BasicReport cmd) {
+	log.debug "rx $cmd"
+	handleDeviceEvent(cmd.value)
+}
+
+def zwaveEvent(sensorbinaryv2.SensorBinaryReport cmd) {
+	log.debug "rx $cmd"
+	def result = []
+
+	if (cmd.sensorType == sensorbinaryv2.SensorBinaryReport.SENSOR_TYPE_TAMPER) {
+		result << createEvent(name: "tamper", value: "detected")
+	} else {
+		result = handleDeviceEvent(cmd.sensorValue)
+	}
+	result
+}
+
+def zwaveEvent(notificationv3.NotificationReport cmd) {
+	def result = []
+
+	log.debug "rx $cmd"
+
+	if (cmd.notificationType == notificationv3.NotificationReport.NOTIFICATION_TYPE_BURGLAR) {
+		if (cmd.event == 3) {
+			result << createEvent(name: "tamper", value: "detected")
+		}
+	} else {
+		log.warn "Unknown cmd.notificationType: ${cmd.notificationType}"
+		result << createEvent(descriptionText: cmd.toString(), isStateChange: false)
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Unhandled Z-Wave command $cmd"
+	createEvent(displayed: false, descriptionText: "$device.displayName: $cmd")
+}
+
+def handleDeviceEvent(value) {
+	log.debug "handleDeviceEvent $value"
+	def result = [
+		createEvent([name: "switch", value: value ? "on" : "off", displayed: false]),
+		createEvent([name: "alarm", value: value ? "both" : "off"])
+	]
+
+	// value != 0 doesn't necessarily trigger the below events,
+	// but value == 0 clears them
+	if (!value) {
+		result << createEvent([name: "chime", value: "off"])
+		result << createEvent([name: "tamper", value: "clear"])
+	}
+
+	result
+}
+
+def generateCommand(command) {
+	def sound = (command && soundMap.containsKey(command)) ? soundMap[command] : soundMap[defaultSound]
+	log.debug "Sending $command"
+
+	secure(zwave.notificationV3.notificationReport(notificationType: sound.notificationType, event: sound.event))
+}
+
+def chime() {
+	log.debug "chime!"
+
+	// Chime is kind of special as the alarm treats it as momentary
+	// and thus sends no updates to us, so we'll send this and request an update
+	sendEvent(name: "chime", value: "chime")
+
+	[
+		generateCommand("Chime"),
+		secure(zwave.basicV1.basicGet())
+	]
+}
+
+def on() {
+	log.debug "sending on"
+	generateCommand(state.sound)
+}
+
+def off() {
+	log.debug "sending off"
+
+	secure(zwave.basicV1.basicSet(value: 0x00))
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
```

Comment:
Should siren be calling `generateCommand("siren")` instead of calling `on`

Reply:
I mean, they'd all be calling the same thing that on is calling. The user configures the siren sound they want, and on, both, strobe, and siren all call that. The chime sound is the only one which behaves completely different. (Basically, if you send a chime and a get immediately the get will always be 0, it is momentary.)

Reply:
well if `state.sound` is not what siren is, would it be playing siren?

Reply:
state.sound is just a populated variable with either the user's selection or the default (since defining something in the preferences block as "default" doesn't necessarily mean settings.<key> gets it until explicitly set). The default is "emergency".

I mean, I suppose I could get really complex and follow the spirit of the packets to a 't' and have on() call BasicSet(0xff) which will always play the "Emergency" sound, and then have siren,strobe,both call generateCommand(state.sound).

Otherwise, I am not sure I understand what you are asking.

Reply:
I followed all of that. The scenario I was describing was that let's say the `state.sound` is set to `Police`. So when `siren()` is called, it would play `Police` sound and maybe that is what the user wants, I was just curious whether the intent is to do `generateCommand("emergency")` or just turn on whatever sound is selected? 

Sorry for the confusion, also I had a typo in my first comment on this.

Reply:
Oh, yeah, the intent is to just use whatever the user has set.

Also, I am going to make a change to test() because that just sends the "Emergency" sound, and that might confuse the user. ("I configured it to Ambulance, pressed test(), but it played Emergency.")

Reply:
Awesome. Sounds good.